tests/network/vmi_multus.go

```diff
@@ -1226,12 +1227,26 @@ var _ = SIGDescribe("[Serial]Macvtap", func() {
 				return vmiIP, nil
 			}
 
+			waitForPodCompleted := func(podNamespace string, podName string) error {
+				pod, err := virtClient.CoreV1().Pods(podNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				for _, containerStatus := range pod.Status.ContainerStatuses {
+					if containerStatus.State.Terminated == nil {
+						return fmt.Errorf("container %s is not terminated, state: %s", containerStatus.Name, containerStatus.State.String())
+					}
+				}
```

Comment:
Would it be possible instead to check the pod phase ? I.e. `succeeded` or `failed` ? It would enable you to not have to check the container statuses.

And, shouldn't you have to check **which** container it is ? for instance, the init container on the pod will *always* be terminated.

Reply:
You're right, checking the pod phase is more straightforward. Thanks

Reply:
Keed in mind that I'm **not** sure you can!

I think it might be possible; and if that is possible, you don't have to concern yourself w/ what container generated the status event. 

If my assumption is wrong - and you actually have to look at the container statuses - you'll **need** to check which container was terminated: only the container holding qemu matters.

Reply:
From [kubernetes doc](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/)

| Pod  phase  | Description |
| ------------- | ------------- |
| Succeeded  | All containers in the Pod have terminated in success, and will not be restarted.  |
| Failed  | All containers in the Pod have terminated, and at least one container has terminated in failure. That is, the container either exited with non-zero status or was terminated by the system.  |

So waiting for either of these is semantically the same as waiting for all containers to terminate.
Not sure if it makes more sense to wait specifically for `compute` container to terminate, in theory that should be the longest lasting container so there shouldn't be a significant difference.

One argument that comes to my mind is that if we document the traffic interruption the way it is proposed in https://github.com/kubevirt/user-guide/pull/468, then this is probably what end users would do as well